client/solve.go

```diff
@@ -423,7 +423,7 @@ func parseCacheOptions(opt SolveOpt) (*cacheOptions, error) {
 					continue
 				}
 				for _, m := range idx.Manifests {
-					if m.Annotations[ocispec.AnnotationRefName] == "latest" {
+					if m.Annotations[ocispec.AnnotationRefName] == "latest" || m.Annotations[ocispec.AnnotationRefName] == attrs["tag"] {
```

Comment:
needs empty check

Reply:
Thank you for the review. Add an empty check.
fixed https://github.com/moby/buildkit/pull/1295/commits/daa2ce2b9859c94ca19d019bbf957ce1c4f40f0b